lib/improver/tests/nbhood/circular_kernel/test_CircularPercentiles.py

```diff
@@ -0,0 +1,396 @@
+        radius = 6000.
+        with self.assertRaisesRegexp(ValueError, msg):
+            CircularPercentiles().run(cube, radius)
+
+    def test_single_point_masked_to_null(self):
+        """Test behaviour with a masked non-zero point.
+
+        The behaviour here is not right, as the mask is ignored.
+        This comes directly from the numpy.percentile base
+        behaviour.
+
+        """
+        cube = set_up_cube()
+        mask = np.zeros_like(cube.data)
+        mask[0, 0, 7, 7] = 1
+        cube.data = np.ma.masked_array(cube.data, mask=mask)
+        expected = np.ones([15] + list(np.shape(cube.data)))
+        expected = np.transpose(expected, [1, 0, 2, 3, 4])
+        expected[0, :, 0, 7, 5] = PERCENTILES_1_IN_13
+        expected[0, :, 0, 6:9, 6] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 5:10, 7] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (5, 1))))
+        expected[0, :, 0, 6:9, 8] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 7, 9] = PERCENTILES_1_IN_13
+        radius = 4000.
+        result = (CircularPercentiles().run(cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_masked_other_point(self):
+        """Test behaviour with a non-zero point next to a masked point.
+
+        The behaviour here is not right, as the mask is ignored.
+
+        """
+        cube = set_up_cube()
+        mask = np.zeros_like(cube.data)
+        mask[0, 0, 6, 7] = 1
+        cube.data = np.ma.masked_array(cube.data, mask=mask)
+        expected = np.ones([15] + list(np.shape(cube.data)))
+        expected = np.transpose(expected, [1, 0, 2, 3, 4])
+        expected[0, :, 0, 7, 5] = PERCENTILES_1_IN_13
+        expected[0, :, 0, 6:9, 6] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 5:10, 7] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (5, 1))))
+        expected[0, :, 0, 6:9, 8] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_13, (3, 1))))
+        expected[0, :, 0, 7, 9] = PERCENTILES_1_IN_13
+        radius = 4000.
+        result = (CircularPercentiles().run(cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_range_1(self):
+        """Test behaviour with a non-zero point with unit range."""
+        cube = set_up_cube()
+        expected = np.ones([15] + list(np.shape(cube.data)))
+        expected = np.transpose(expected, [1, 0, 2, 3, 4])
+        expected[0, :, 0, 7, 6] = PERCENTILES_1_IN_5
+        expected[0, :, 0, 6:9, 7] = (
+            np.transpose(np.tile(PERCENTILES_1_IN_5, (3, 1))))
+        expected[0, :, 0, 7, 8] = PERCENTILES_1_IN_5
+        radius = 2000.
+        result = (CircularPercentiles().run(cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_range_0(self):
```

Comment:
Shouldn't this behaviour be tested somewhere else? It is in test_spatial, maybe we need to look at the overlap here?